frontend/pages/product_new_page.js

```diff
@@ -77,6 +81,7 @@ export default class AddProductForm extends PureComponent {
 
   addProduct = (e) => {
     // e.preventDefault();
+    if (!this.state.dataInputStatus) this.setState({ dataInputStatus: 'incomplete' });
```

Comment:
setState is async, so doing `this.setState` here is most likely not going to update the state before the next line. I'd make it like this instead

```
const payload = {
  product: {
    ...this.state,
    ingredients,
    dataInputStatus: this.state.dataInputStatus || 'incomplete',
  },
};
http.post('/product', payload).then ...
```